### PR TITLE
[DE-574] Java Driver v7

### PR DESCRIPTION
**Pull request by rashtao**

### CODE REVIEW

File: src/main/java/com/arangodb/springframework/core/convert/ArangoEntityWriter.java [hunk 1]
```diff
@@ -20,22 +20,15 @@
...
  *
  */
-public interface ArangoEntityWriter extends EntityWriter<Object, VPackBuilder> {
-
-	default VPackSlice write(final Object source) {
-		final VPackBuilder builder = new VPackBuilder();
-		write(source, builder);
-		return builder.slice();
-	}
+public interface ArangoEntityWriter {
```
Comment (aburmeis) [inline src/main/java/com/arangodb/springframework/core/convert/ArangoEntityWriter.java:31]: why no longer implement `EntityWriter`?

File: src/main/java/com/arangodb/springframework/core/template/ArangoExtCursor.java [hunk 2]
```diff
@@ -20,38 +20,124 @@
...
+    }
+
+    @Override
+    public boolean isCached() {
+        return delegate.isCached();
+    }
+
+    @Override
+    public List<T> asListRemaining() {
+        return delegate.asListRemaining().stream()
```
Comment (aburmeis) [inline src/main/java/com/arangodb/springframework/core/template/ArangoExtCursor.java:90]: why not simply `delegate.stream()`?

**Review by aburmeis — APPROVED** [review]

**Comment by aburmeis** [comment]
As Driver 7 has braking changes I would recommend to release this together with Spring Boot 3 support!

File: src/main/java/com/arangodb/springframework/core/convert/ArangoEntityWriter.java (hunk 1 above)
Comment (rashtao) [reply src/main/java/com/arangodb/springframework/core/convert/ArangoEntityWriter.java:31]: EntityWriter is an `Interface to write objects into store specific sinks`. But here we have no need to append data invoking `write()` mutiple time, instead the method can directly return the final `JsonNode`.
IMHO the use we made of it before was unneeded:
```
public interface ArangoEntityWriter extends EntityWriter<Object, VPackBuilder> {

	default VPackSlice write(final Object source) {
		final VPackBuilder builder = new VPackBuilder();
		write(source, builder);		// <-- here is the only place where EntityWriter.write() is invoked
		return builder.slice();
	}

}
```

**Comment by rashtao** [comment]
Yes, also [ArangoDB spring-boot-starter](https://github.com/arangodb/spring-boot-starter) needs to be updated.

**Pull request merged**